src/main/java/org/example/pages/LoginPage.java

```diff
@@ -1,11 +1,55 @@
 package org.example.pages;
 
-import org.openqa.selenium.By;
-import org.openqa.selenium.WebDriver;
+import java.util.List;
 
-public class LoginPage extends HomePage {
+import org.openqa.selenium.*;
+import org.openqa.selenium.interactions.Actions;
 
-	public LoginPage(WebDriver driver) {
-		super(driver);
-	}
+/**
+ * LoginPage is the initial page of the application.
+ *
+ */
+public class LoginPage {
+    private WebDriver driver;
+
+    public LoginPage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public WebElement findElement(By locator) {
+        return driver.findElement(locator);
+    }
+
+    public List<WebElement> findElements(By locator) {
+        return driver.findElements(locator);
+    }
+
+    public String getText(WebElement element) {
+        return element.getText();
+    }
+
+    public String getText(By locator) {
+        return driver.findElement(locator).getText();
+    }
+
+    public void type(String inputText, By locator) {
+        driver.findElement(locator).sendKeys(inputText);
+    }
+
+    public void click(By locator) {
+        driver.findElement(locator).click();
+    }
+
+    public void hoverElement(By locator) {
+        Actions actions = new Actions(driver);
+        actions.moveToElement(findElement(locator)).perform();
+    }
+
+    public Boolean isDisplayed(By locator) {
+        try {
+            return driver.findElement(locator).isDisplayed();
+        } catch (org.openqa.selenium.NoSuchElementException e) {
+            return false;
+        }
+    }
```

Comment:
Please create the class that will be a parent class (Let's say 'MainFunctional' or 'BasePage') for all future pages includes LoginPage class and move all those method into that class.
Why it needs, currently you have a lot of basic functional to work with web elements and when you will add a new Pages (as required by PageObject pattern, you will be forced to duplicate those methods). With parent classs approach all you need it just inheritance from BasePage (or whatever you named it) and you all set and ready to move forward.

Reply:
Did It, MainFunctional Class created